tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -1558,6 +1559,38 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
 		oa.UpgradeTidbClusterOrDie(&tcCfg)
 		oa.CheckTidbClusterStatusOrDie(&tcCfg)
 	})
+
+	ginkgo.It("Deleted objects controlled by TidbCluster will be recovered by Operator", func() {
+		ginkgo.By("Deploy initial tc")
+		tc := fixture.GetTidbCluster(ns, "delete-objects", utilimage.TiDBV4)
+		utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc, 5*time.Minute, 10*time.Second)
+
+		ginkgo.By("Delete StatefulSet/ConfigMap/Service of PD")
+		deleteResourcesForComponent := func(memberName string) {
```

Comment:
What about using label selector `app.kubernetes.io/instance` to delete the resources without caring about the components?